packages/flutter/lib/src/cupertino/magnifier.dart

```diff
@@ -0,0 +1,305 @@
+  ///
+  /// The default constructor parameters and constants were eyeballed on
+  /// an iPhone XR iOS v15.5.
+  const CupertinoMagnifier({
+    super.key,
+    this.size = kDefaultSize,
+    this.borderRadius = const BorderRadius.all(Radius.elliptical(60, 50)),
+    this.additionalFocalPointOffset = Offset.zero,
+    this.borderSide =
+        const BorderSide(color: Color.fromARGB(255, 235, 235, 235)),
+    this.inOutAnimation,
+  });
+  // These constants were eyeballed on an iPhone XR iOS v15.5.
+
+  /// The border, or "rim", of this magnifier.
+  final BorderSide borderSide;
+
+  /// The vertical offset, that the magnifier is along the Y axis above
+  /// the focal point.
+  @visibleForTesting
+  static const double kMagnifierAboveFocalPoint = -25;
+
+  /// The default size of the magnifier.
+  ///
+  /// This is public so that positioners can choose to depend on it, although
+  /// it is overrideable.
+  @visibleForTesting
+  static const Size kDefaultSize = Size(82.5, 45);
+
+  /// The duration that this magnifier animates in / out for.
+  ///
+  /// The animation is a translation and a fade. The translation
+  /// begins at the focal point, and ends at [kMagnifierAboveFocalPoint].
+  /// The opacity begins at 0 and ends at 1.
+  static const Duration _kInOutAnimationDuration = Duration(milliseconds: 150);
+
+  /// The size of this magnifier.
+  final Size size;
+
+  /// The border radius of this magnifier.
+  final BorderRadius borderRadius;
+
+  /// This [RawMagnifier]'s controller.
+  ///
+  /// Since [CupertinoMagnifier] has no knowledge of shown / hidden state,
+  /// this animation should be driven by an external actor.
+  final Animation<double>? inOutAnimation;
+
+  /// Any additional focal point offset, applied over the regular focal
+  /// point offset defined in [kMagnifierAboveFocalPoint].
+  final Offset additionalFocalPointOffset;
+
+  @override
+  Widget build(BuildContext context) {
+    Offset focalPointOffset =
+        Offset(0, kMagnifierAboveFocalPoint - kDefaultSize.height / 2);
+    focalPointOffset.scale(1, inOutAnimation?.value ?? 1);
+    focalPointOffset += additionalFocalPointOffset;
+
+    return Transform.translate(
+        offset: Offset.lerp(
+          const Offset(0, -kMagnifierAboveFocalPoint),
+          Offset.zero,
+          inOutAnimation?.value ?? 1,
+        )!,
+        child: RawMagnifier(
+          size: size,
+          focalPoint: focalPointOffset,
+          decoration: MagnifierDecoration(
+            opacity: inOutAnimation?.value ?? 1,
+            shape: RoundedRectangleBorder(
+              borderRadius: borderRadius,
+              side: borderSide,
+            ),
+            shadows: const <BoxShadow>[
+              BoxShadow(
+                color: Color.fromARGB(34, 0, 0, 0),
+                blurRadius: 5,
+                spreadRadius: 0.2,
+                offset: Offset(0, 3),
+              )
+            ],
+          ),
+        ));
```

Comment:
Nit: I think these two parentheses might look better on their own lines, and a trailing comma after the first.